engine/collection/compliance/core.go

```diff
@@ -251,7 +251,15 @@ func (c *Core) processBlockAndDescendants(proposal *cluster.Block) error {
 		if checkForAndLogOutdatedInputError(err, log) || checkForAndLogUnverifiableInputError(err, log) {
 			return nil
 		}
-		if checkForAndLogInvalidInputError(err, log) {
+		if invalidBlockErr, ok := model.AsInvalidProposalError(err); ok {
+			log.Err(err).Msg("received invalid block from other node (potential slashing evidence?)")
+
+			// notify consumers about invalid block
+			c.proposalViolationNotifier.OnInvalidBlockDetected(flow.Slashable[model.InvalidProposalError]{
+				OriginID: originID,
+				Message:  *invalidBlockErr,
+			})
```

Comment:
## ⚠️ Potentially faulty attribution of protocol violation (?)

I think we could be potentially accusing the wrong party of having given us an invalid proposal. This is because the `originID` here is recursively passed down from the [`OnBlockProposal(originID, proposal)` call](https://github.com/onflow/flow-go/blob/339fcc04b1575c86f4c092780167c009d44469a7/engine/collection/compliance/core.go#L114) that _initiated the processing of this fork. 

For example consider the following fork
* `A` <- `B` <- `C` (these are all honest blocks)
* This replica has block `B` already but is still missing block `C`
* A malicious node is giving us block a _broken_ block `X`  with parent `C`
   * <- `B` <- ?unknown block C? <- `X`(broken)
* we now receive `C` from an honest node. Hence we proceed to extend the protocol state with `X`
   * processing  `C` here works as expected https://github.com/onflow/flow-go/blob/339fcc04b1575c86f4c092780167c009d44469a7/engine/collection/compliance/core.go#L226 
   * but in the recursive call for processing `C`'s children, we just pass along `originID` https://github.com/onflow/flow-go/blob/339fcc04b1575c86f4c092780167c009d44469a7/engine/collection/compliance/core.go#L286
     i.e. we blame the node that gave us `C` for invalidity of any children, even if we received the children from other nodes